src/pasta.rs

```diff
@@ -204,14 +205,20 @@ impl Group for pallas::Point {
     let mut shake = Shake256::default();
     shake.input(label);
     let mut reader = shake.xof_result();
-    let mut gens: Vec<Self::PreprocessedGroupElement> = Vec::new();
-    let mut uniform_bytes = [0u8; 32];
+    let mut uniform_bytes_vec = Vec::new();
     for _ in 0..n {
+      let mut uniform_bytes = [0u8; 32];
       reader.read_exact(&mut uniform_bytes).unwrap();
-      let hash = Ep::hash_to_curve("from_uniform_bytes");
-      gens.push(hash(&uniform_bytes).to_affine());
+      uniform_bytes_vec.push(uniform_bytes);
     }
-    gens
+    (0..n)
+      .collect::<Vec<usize>>()
+      .into_par_iter()
+      .map(|i| {
+        let hash = Ep::hash_to_curve("from_uniform_bytes");
+        hash(&uniform_bytes_vec[i]).to_affine()
```

Comment:
Was just looking at doing this myself since the [zator](https://github.com/lyronctk/zator) folks mentioned slow performance here:

One other minor optimisation you can make here is to use [`batch_normalize`](https://docs.rs/pasta_curves/latest/pasta_curves/struct.Ep.html#method.batch_normalize) for this conversion. One clean way to do both optimisations is to do the rayon parallelisation manually by splicing the list into `num_threads` pieces